panicwrap.go

```diff
@@ -274,7 +278,7 @@ func trackPanic(r io.Reader, w io.Writer, dur time.Duration, result chan<- strin
 			// look for another panic along the way.
 
 			// First, remove the previous panic header so we don't loop
-			w.Write(panicBuf.Next(len(panicHeader)))
+			w.Write(panicBuf.Next(len(panicHeaders[panicType])))
```

Comment:
I'm struggling to follow all the logic in this section, but it looks possible that panicType could be -1 at this point, and attempting to get the correct header might fail.

even if that's not the case then a small change in the future might make this the case, so it might be worth adding an extra check here.

Reply:
Yeah its not the easiest to follow - I'll tweak it a bit to make it more straightforward and rule out the edge case.